sketch/dim_reduce.h

```diff
@@ -13,11 +13,10 @@ class Int32Flattener {
   public:
     using sketch_type = std::vector<uint32_t>;
 
-    Int32Flattener(uint32_t flat_dim, uint32_t sketch_dim, uint32_t max_len)
+    Int32Flattener(uint32_t flat_dim, uint32_t sketch_dim, uint32_t max_len, unsigned int seed)
```

Comment:
We typically use platform independent ints where possible. Here I see that rd() actually returns an `unsigned int`, so perhaps it makes sense, but I think I'd prefer a uint32_t. You could also use `std::random_device::result_type` but that's probably an overkill?

Reply:
Oh yes, I wanted to write about this:
https://en.cppreference.com/w/cpp/numeric/random/random_device 'hardcodes' that `result_type` to `unsigned int`, but also happy to write `uint32_t` everywhere